src/main/scala/com/brigade/finagle/consul/catalog/ConsulAgentClient.scala

```diff
@@ -0,0 +1,102 @@
+package com.brigade.finagle.consul.catalog
+
+import com.brigade.finagle.consul.ConsulQuery
+import com.twitter.finagle.Service
+import com.twitter.finagle.http.{Method, Request, Response}
+import com.twitter.logging.Logger
+import com.twitter.util.{Duration, Future}
+import org.json4s.DefaultFormats
+import org.json4s.jackson.Serialization
+
+import java.net.InetSocketAddress
+import java.util.concurrent.TimeUnit
+
+/**
+ * A wrapper around Consul agent service registration HTTP calls
+ */
+class ConsulAgentClient(httpClient: Service[Request, Response]) {
+
+  import ConsulAgentClient._
+
+  private lazy val log = Logger.get(getClass)
+  implicit val formats = DefaultFormats
+
+  def healthCheckPath(checkId: String): String = s"/v1/agent/check/pass/$checkId"
+
+  def registerPath: String = "/v1/agent/service/register"
+
+  def deregisterPath(serviceId: String): String = s"/v1/agent/service/deregister/$serviceId"
+
+  def formatTtl(d: Duration): String = s"${d.inUnit(TimeUnit.SECONDS)}s"
+
+  def register(ia: InetSocketAddress, q: ConsulQuery): Future[RegisterResponse] = {
+    val address = ia.getAddress.getHostAddress
+    val port = ia.getPort
+
+    // use the service name + the address to allow multiple instances on the same host
+    val serviceId = s"${q.name}-$address-$port"
+
+    // See: https://consul.io/docs/agent/services.html, the check id is set to service:service-id
+    val checkId = s"service:$serviceId"
+
+    val check = TtlCheck(TTL = formatTtl(q.ttl))
+    val serviceDefJson = ConsulServiceJson(
+      ID = Some(serviceId),
```

Comment:
Not sure what you are asking, this is never `None` here. ID is optional for consul